hack/chains/end-to-end-demo.sh

```diff
@@ -0,0 +1,138 @@
+and token make sure that it's allowed to push to ${APP_IMAGE_NAME} repository.
+"
+  docker login quay.io --username "${MY_QUAY_USER}"
+fi
+
+# Making the quay.io repository public, this is a requirement right now for the Test stream tasks to succeed
+if [[ "$(curl -s "https://quay.io/api/v1/repository?namespace=${MY_QUAY_USER}&public=true" | jq ".repositories[] | select(.name==\"${APP_IMAGE_NAME}\") | has(\"name\")")" != 'true' ]]; then
+  echo "🛑 Make sure that the quay.io repository '${APP_IMAGE_NAME}' exist and it's visibility
+is set to public, create an empty repository here:
+
+👉 https://quay.io/new/?name=${APP_IMAGE_NAME}
+
+if the repository already exists change the visibility here:
+
+👉 https://quay.io/repository/${MY_QUAY_USER}/${APP_IMAGE_NAME}?tab=settings "
+  exit 1
+fi
+
+# We're going to assume running with kubeadmin
+if [[ "$(kubectl auth can-i '*' '*' < /dev/null 2>&1)" != 'yes' ]]; then
+  echo "🛑 Make sure that you're logged in as kubeadmin, we make assumptions here"
+  exit 1
+fi
+
+# Checking the status of the build component as a way to make sure that the cluster is okay to run the demo
+if [[ "$(kubectl get -n openshift-gitops applications.argoproj.io build -o jsonpath='{.status.sync.status}')" != 'Synced' ]]; then
+  echo "🛑 Make sure that the cluster is setup by running hack/bootstrap-cluster.sh first see
+👉 https://github.com/redhat-appstudio/infra-deployments#bootstrapping-a-cluster"
+  exit 1
+fi
+
+echo "📜 The outline of the demo
+  1. We'll start a build pipeline using build-definitions/hack/test-build.sh
+     to build a Node.js app. This will include running the sanity checks from
+     the Test stream and gathering attestations and signatures from the
+     Contract team.
+  2. Next, a sample release pipeline will be run that will check the results
+     from the build pipeline and copy the image to the production environment.
+
+We'll show the commands being run, and point to data or visualisations.
+"
+
+HACK_CHAINS_DIR="$(dirname "$0")"
+
+BUILD_DEFINITIONS_DIR="${HACK_CHAINS_DIR}/../../../build-definitions"
+if [[ ! -d "${BUILD_DEFINITIONS_DIR}" ]]; then
+  echo "ℹ️  Did not find redhat-appstudio/build-definitions clone in ${BUILD_DEFINITIONS_DIR}, using upstream version"
+  BUILD_DEFINITIONS_DIR="$(mktemp -d build-definitions.XXXXXXXXXX)"
+  trap "rm -rf ${BUILD_DEFINITIONS_DIR}" EXIT
+  (cd "${BUILD_DEFINITIONS_DIR}" && curl -sLO https://github.com/redhat-appstudio/build-definitions/archive/refs/heads/main.zip && unzip main.zip && mv build-definitions-main/* . && rm main.zip)
+fi
+
+{
+  oc project demo 2>/dev/null 1>&2 && echo -n "ℹ️  Using existing 'demo' OpenShift project"
+} || {
+  echo "📂 Creating a new OpenShift project 'demo'
+"
+  oc new-project demo
+}
+echo "Here is the OpenShift console with the project:
+
+👉 $(oc whoami --show-console)/k8s/cluster/projects/demo
+"
+
+echo "♾️  Setting up pipelines
```

Comment:
Is the two spaces vs one space intentional here, (and in a few other places)?

Reply:
Could be my shell/terminal/font but unless I do that there is no space between the emoji and text. I can see two spaces in http://pastebin.test.redhat.com/1047960, so I'll revert to using one